### ISSUE TITLE
[Bug]: Pipeline server fails when using object storage "signed by unknown authority"

### ISSUE THREAD

**Issue by erwangranger** [opendatahub-io/data-science-pipelines-operator#362]
### ODH Component

Data Science Pipelines

### Current Behavior

Reporting this for a customer I am working with. 

The pipeline pod fails with this message: 

```
bash-3.2$ oc logs -f ds-pipeline-pipelines-definition-6cd6859c6f-twmxm -n test-model-serving
Defaulted container "ds-pipeline-api-server" out of: ds-pipeline-api-server, oauth-proxy
I0914 20:42:34.297035       1 client_manager.go:160] Initializing client manager
I0914 20:42:34.297126       1 config.go:74] Config DBConfig.ExtraParams not specified, skipping
F0914 20:42:34.449271       1 client_manager.go:411] Failed to check if Minio bucket exists. Error: Get https://my.int.s3.corporate.int:443/zone-1-dev/?location=: x509: certificate signed by unknown authority
``` 

### Expected Behavior

I should be able to use an Object Storage even if it is using a self-signed cert. 

### Workaround (if any)

None found

### Open Data Hub Version

```yml
RHODS 1.31
```

### Anything else

This is happening in a disconnected environment.

**Comment by shalberd** [comment]
minor added note: our solution in notebooks is for acessing secured Routes with either self-signed certificates or custom PKI issued certificates only, not for .svc that are potentially encryped cluster-internally. That is ok, though. Securing services cluster-internally is a whole different ballgame.

**Comment by gregsheremeta** [comment]
here's a document that describes a workaround for this while we work on an official fix

https://docs.google.com/document/d/15Fj-l0xDXXJAraZMXkT-SDD12BuYe-KfHITWCmtl-rQ/edit?usp=sharing

**Comment by HumairAK** [comment]
Note that the resolution to this has only been implemented backend side for rhods-2.5, there is still UI yet to be added for this fix, tracking issue is [here](https://github.com/opendatahub-io/odh-dashboard/issues/1711). This can be used from 2.5 to simplify a work around.

Workaround: 
---
After starting a Pipeline Server with unrecognized certs, it will fail to start (the logs [currently only get surfaced in the DSPO logs](https://github.com/opendatahub-io/data-science-pipelines-operator/issues/451)) 

Upload a configmap with the ca cert bundle in PEM format, for example: 

```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: custom-ca-bundle
data:
  ca-bundle.crt: |
    # contents of ca-bundle.crt
```

Then patch the underlying `DataSciencePipelinesApplicaitonResource` like so: 

```bash
DSPA_NAME=ds-pipeline-pipelines-definition
DSPA_NAMESPACE=your-ds-project

patch='{"spec":{"apiServer":{"cABundle":{"configMapKey":"ca-bundle.crt","configMapName":"custom-ca-bundle"}}}}'
oc patch dspa ${DSPA_NAME} -n ${DSPA_NAMESPACE}  --type=merge -p ${patch}
```

Pipeline Server should come up successfully.
